src/control/control.h

```diff
@@ -13,6 +13,17 @@
 
 class ControlObject;
 
+enum class ControlFlag {
+    None = 0,
+    InitializeLater = 1,
```

Comment:
I have thought about the "InitializeLater" flag and I think we can do better. 
Let's either remove this and assert or make early initialization work. 
In any case we should remove the flag otherwise I am afraid this will be a source of a mayor confusion for new developers.  

See my other comments for fix the root issue.